bertopic/_utils.py

```diff
@@ -147,3 +147,57 @@ def validate_distance_matrix(X, n_samples):
         raise ValueError("Distance matrix cannot contain negative values.")
 
     return X
+
+
+def select_topic_representation(
+    ctfidf_embeddings: Optional[Union[np.ndarray, csr_matrix]] = None,
+    semantic_embeddings: Optional[np.ndarray] = None,
+    use_ctfidf: bool = True
+) -> tuple[np.ndarray, bool]:
+    """Select the topic representation.
+
+    Arguments:
+        ctfidf_embeddings: The c-TF-IDF embedding matrix
+        semantic_embeddings: The semantic embedding matrix
+        use_ctfidf: Whether to use the c-TF-IDF representation. If False, topics embedding representation is used, if it
+                    exists. Default is True.
+    Raises
+        ValueError:
+            - If no topic representation was found
+            - If c-TF-IDF embeddings are not a numpy array or a scipy.sparse.csr_matrix
+
+    Returns:
+        The selected topic representation and a boolean indicating whether it is c-TF-IDF.
+    """
+
+    def to_ndarray(array: Union[np.ndarray, csr_matrix]) -> np.ndarray:
+        if isinstance(array, csr_matrix):
+            return array.toarray()
+        elif not isinstance(array, np.ndarray):
+            raise ValueError("The embeddings should be either a type of `numpy.ndarray` or `scipy.sparse.csr_matrix`")
```

Comment:
Why are you checking for this? The input is always either np.array or csr_matrix, so this should not happen in the first place right?

Reply:
Due to the code readability. Both embeddings can be `None` (by function types). This is the way how to say that one must not be None. 

Not a big deal, I can delete it if you like.

Reply:
It depends who is calling this function, no? 

I can delete it though if you like.

Reply:
> It depends who is calling this function, no?

What do you mean? Since this is a private function, nobody should be using in this in the first place right? Or is there some other usage where this might be happening?

Reply:
In my view, it would be less readable without the check

Here are two [best practices](https://betterprogramming.pub/clean-code-and-the-practice-of-programming-for-newcomers-2dfaeb0e31c3) that it breaks : 
Write for readability — code is read more often than it’s written.
Explicit is better than implicit.

Anyway, I will remove it.

Reply:
Thanks for sharing! Do note though that I am not a newcomer as the blog post suggests 😉

I understand your view but do note that these best practices that you mention are very broad and can be interpreted from many perspectives. For instance, the code to me is less readable and implicitly communicates that we can certain situations arise from general use. 

For now, let's remove it considering the way the rest of the codebase follows a different style.

Reply:
I did not mean to suggest that you are a newcomer, sorry. I did not notice the title. I was looking for a reference to my approach. 

It is removed.